readthedocs/api/v2/views/model_views.py

```diff
@@ -224,11 +225,25 @@ class VersionViewSet(DisableListEndpoint, UserSelectViewSet):
 class BuildViewSet(DisableListEndpoint, UserSelectViewSet):
     permission_classes = [APIRestrictedPermission]
     renderer_classes = (JSONRenderer, PlainTextBuildRenderer)
-    serializer_class = BuildSerializer
-    admin_serializer_class = BuildAdminSerializer
     model = Build
     filterset_fields = ('project__slug', 'commit')
 
+    def get_serializer_class(self):
+        """
+        Return the proper serializer for UI and Admin.
+
+        This ViewSet has a sligtly different pattern since we want to
+        pre-process the `command` field before returning it to the user, and we
+        also want to have a specific serializer for admins.
+        """
+        if self.request.user.is_staff:
+            # Logic copied from `UserSelectViewSet.get_serializer_class`
+            # and extended to check for GET method
+            if self.action in ["list", "retrieve"]:
+                return BuildAdminReadOnlySerializer  # Staff read-onlyl
+            return BuildAdminSerializer  # Staff write-only
```

Comment:
```suggestion
            # Logic copied from `UserSelectViewSet.get_serializer_class`
            # and extended to choose serializer from self.action
            if self.action not in ["list", "retrieve"]:
                return BuildAdminSerializer  # Staff write-only
            return BuildAdminReadOnlySerializer  # Staff read-only
```

This is a really small thing, but since the `BuildAdminSerializer` is the one with the elevated permissions, the conditions are better to set explicitly.

(but `self.request.user.is_staff` does the job very well otherwise)